mesheryctl/internal/cli/root/system/context/view.go

```diff
@@ -23,8 +25,22 @@ var viewContextCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
+		//Storing all the tokens separately in a map, to get tokenlocation by token name.
+		for _, tok := range configuration.Tokens {
+			tokenNameLocation[tok.Name] = tok.Location
+		}
 
 		if allContext {
+			for k, v := range configuration.Contexts {
+				if v.Token == "" {
+					log.Warnf("[Warning]: Token not specified/empty for context \"%s\"", k)
+					v.TokenLocation = ""
+				} else {
+					addTokenLocation(&v, k)
```

Comment:
So instead of adding the location to the struct, we can directly print the config.Token with the name that has been stored in a map[string][config.Token]

See, the point is why tokens(name, and location) are separated from a context is that are reusable for multiple contexts what you implemented here doesn't align with that.

Reply:
or you could do the same but with a temp struct not with editing the Config.Context struct.

Reply:
@piyushsingariya  # I added an auxilliary struct instead of modifying the original context as you mentioned. 
![1](https://user-images.githubusercontent.com/43276904/125059635-29d7f800-e0c9-11eb-80b0-d8de97efa9eb.png)

## Incase the token field is left empty/or not mentioned in a specific context:
![2](https://user-images.githubusercontent.com/43276904/125061534-1168dd00-e0cb-11eb-9f12-a86d4cf87eda.png)

## Incase the token is not present inside the tokens: array
![3](https://user-images.githubusercontent.com/43276904/125061718-3f4e2180-e0cb-11eb-93b5-9cbf7ed22cd0.png)

## When both cases individually in different contexts:
![4](https://user-images.githubusercontent.com/43276904/125061938-7ae8eb80-e0cb-11eb-94ec-0ee86f6e5342.png)

Reply:
I love seeing these test cases. Are these covered by unit tests, too?

Reply:
I don't think there are any test cases for any of the functions inside of context sub-command or even "meshery system" command in general as of now.  It has been pointed out in issue #3407

Reply:
Thank you, @revolyssup. Batter-up, in that case. 🙂 

Note the Acceptance Tests for the https://github.com/meshery/meshery/issues/3401 issue.

Reply:
@leecalcote  For unit tests, currently there is a single test yaml inside pkg/utils, that is being used in "system channel" test.
I can write unit tests for my specific changes in "system context view", but I would also need some test yaml files for covering different cases.
How do you propose I move forward?
1. Concatenate all 4 test cases in a single new test yaml in pkg/utils and use that for my tests. (Cannot concatenate in already existing yaml as that yaml is being parsed in existing test as a struct and not slice of structs)
2. Create a sub directory inside pkg/utils and use some descriptive naming convention to create separate yaml files, as there can be a need of more test YAMLs to cover different cases in different commands and subcommands.

Reply:
+ @hexxdump

Reply:
Wait for this weekend, I'll raise a PR that would help you set up the context with custom path. and you should cover that in a separate PR tbh.